src/h2n_fetcher.erl

```diff
@@ -0,0 +1,396 @@
+
+-spec decorate_app(hex2nix:deps(), {hex2nix:app(), [hex2nix:app()]}) ->
+                          dep_desc().
+decorate_app(#indexed_deps{roots=Roots},
+             {App={AppName, AppVsn}, Deps}) ->
+    IsRoot = case sets:is_element(App, Roots) of
+                 true ->
+                     root;
+                 false ->
+                     historical
+             end,
+    {Description, Licenses, Link} =
+        get_metadata(AppName, get_app_detail_from_hex_pm(AppName)),
+    {Sha, HasNativeCode} = get_deep_meta_for_package(AppName, AppVsn),
+    #dep_desc{app = App
+             , description = Description
+             , position = IsRoot
+             , licenses = Licenses
+             , homepage = Link
+             , sha = Sha
+             , has_native_code = HasNativeCode
+             , deps = Deps}.
+
+-spec get_app_detail_from_hex_pm(binary()) -> jsx:json_term().
+get_app_detail_from_hex_pm(AppName) ->
+    Thunk =
+        fun() ->
+                Url = h2n_util:iolist_to_list(["https://hex.pm/api/packages/"
+                                              , AppName]),
+                ibrowse:send_req(Url
+                                , [{"Accept", "application/json"}]
+                                , get
+                                , []
+                                , [])
+        end,
+    {ok, "200", _, Body} =
+        take_at_least_one_second(Thunk),
+    jsx:decode(erlang:iolist_to_binary(Body)).
+
+-spec get_deep_meta_for_package(hex2nix:app_name(), hex2nix:app_version()) ->
+                                 {sha(), boolean()}.
+get_deep_meta_for_package(AppName, AppVsn) ->
+    TempDirectory = h2n_util:temp_directory(),
+    Package = binary_to_list(<<AppName/binary, "-", AppVsn/binary, ".tar">>),
+    TargetPath = filename:join(TempDirectory, Package),
+    Url = h2n_util:iolist_to_list([?DEFAULT_CDN, "/", Package]),
+    io:format("Pulling From ~s to ~s~n"
+             , [Url, TargetPath]),
+    Thunk =
+        fun() ->
+                ibrowse:send_req(Url
+                                , []
+                                , get
+                                , []
+                                , [{save_response_to_file,
+                                    TargetPath}])
+        end,
+    {ok, "200", _, {file, DownloadedPath}} =
+        take_at_least_one_second(Thunk),
+    [Sha | _]  = string:tokens(os:cmd(io_lib:format("sha256sum \"~s\"", [DownloadedPath])),
+                               " "),
+    io:format("Got Sha ~s for ~s~n", [Sha, Package]),
+    {erlang:list_to_binary(Sha), has_native_code(TempDirectory, TargetPath)}.
+
+-spec has_native_code(file:filename(), file:filename()) -> boolean().
+has_native_code(TempDirectory, TargetPath) ->
+    ok = erl_tar:extract(TargetPath, [{cwd, TempDirectory}]),
+    ContentsPath = filename:join(TempDirectory, "contents.tar.gz"),
+    DirListing =
+        string:tokens(os:cmd(io_lib:format(" tar tf ~s", [ContentsPath])), "\n"),
```

Comment:
That doesn't work because the original tar is a tarball of metadata and 'contents.tar.gz'. Unfortunately, you can't get the table of a tar inside another tar :/